tests/by-util/test_sleep.rs

```diff
@@ -129,29 +131,95 @@ fn test_sleep_wrong_time() {
 }
 
-// TODO These tests would obviously block for a very long time. We
-// only want to verify that there is no error here, so we could just
-// figure out a way to terminate the child process after a short
-// period of time.
+#[test]
+fn test_sleep_when_single_input_exceeds_max_duration_then_no_error() {
+    let mut child = new_ucmd!()
+        .arg(format!("{}", u64::MAX as u128 + 1))
+        .timeout(Duration::from_secs(10))
+        .run_no_wait();
+
+    #[cfg(unix)]
+    child
+        .delay(100)
+        .kill()
+        .make_assertion()
+        .with_current_output()
+        .signal_is(9) // make sure it was us who terminated the process
+        .no_output();
+    #[cfg(windows)]
+    child
+        .delay(100)
+        .kill()
+        .make_assertion()
+        .with_current_output()
+        .failure()
+        .no_output();
+}
 
-// #[test]
-#[allow(dead_code)]
-fn test_dont_overflow() {
+#[test]
+fn test_sleep_when_multiple_inputs_exceed_max_duration_then_no_error() {
+    let mut child = new_ucmd!()
+        .arg(format!("{}", u64::MAX))
+        .arg("1")
+        .timeout(Duration::from_secs(10))
+        .run_no_wait();
+
+    #[cfg(unix)]
+    child
+        .delay(100)
+        .kill()
+        .make_assertion()
+        .with_current_output()
+        .signal_is(9) // make sure it was us who terminated the process
+        .no_output();
+    #[cfg(windows)]
+    child
+        .delay(100)
+        .kill()
+        .make_assertion()
+        .with_current_output()
+        .failure()
+        .no_output();
+}
+
+#[rstest]
+#[case::whitespace_prefix(" 0.1s")]
+#[case::multiple_whitespace_prefix("   0.1s")]
+#[case::whitespace_suffix("0.1s ")]
+#[case::mixed_newlines_spaces_tabs("\n\t0.1s \n ")]
+fn test_sleep_when_input_has_whitespace_then_no_error(#[case] input: &str) {
     new_ucmd!()
-        .arg("9223372036854775808d")
+        .arg(input)
+        .timeout(Duration::from_secs(10))
         .succeeds()
-        .no_stderr()
-        .no_stdout();
+        .no_output();
 }
 
-// #[test]
-#[allow(dead_code)]
-fn test_sum_overflow() {
+#[rstest]
+#[case::only_space(" ")]
+#[case::only_tab("\t")]
+#[case::only_newline("\n")]
+fn test_sleep_when_input_has_only_whitespace_then_error(#[case] input: &str) {
     new_ucmd!()
-        .args(&["100000000000000d", "100000000000000d", "100000000000000d"])
-        .succeeds()
-        .no_stderr()
-        .no_stdout();
+        .arg(input)
+        .timeout(Duration::from_secs(10))
+        .fails()
+        .usage_error(format!("invalid time interval '{input}'"));
+}
+
+#[test]
+fn test_sleep_when_multiple_input_some_with_error_then_shows_all_errors() {
+    let expected = "invalid time interval 'abc'\n\
+        sleep: invalid time interval '1years'\n\
+        sleep: invalid time interval ' '";
+
+    // We also test here with a high sleep interval if the only correct interval is applied or not
+    // by setting the timeout of this run to something much lower.
```

Comment:
I don't think I understand this comment? The test makes sense, but timeout shouldn't matter right, because it shouldn't run anyway?

Reply:
It's about the hypothetical case, that the errors are printed but `sleep` calls `thread::sleep` with the sleep interval of the only correct number (`100000.0`). The default timeout of `30s` would have been sufficient but I'm setting the timeout explicitly here, so it is more obvious that the test fails if sleep is running for more than `10s` and also makes the timeout an integral part of this test run.

Reply:
Hmm I see. What about this phrasing then:
```rust
// Even if one of the arguments is valid, but the rest isn't, we should still fail and exit early.
// So, we set a timeout of 10 seconds to ensure that we don't sleep for 10000 seconds.
```